backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotsResource.java

```diff
@@ -36,13 +36,18 @@ public DiskSnapshots list() {
     protected DiskSnapshots mapCollection(List<org.ovirt.engine.core.common.businessentities.storage.Disk> entities) {
         DiskSnapshots collection = new DiskSnapshots();
         for (org.ovirt.engine.core.common.businessentities.storage.Disk disk : entities) {
-            DiskSnapshot diskSnapshot = getMapper(org.ovirt.engine.core.common.businessentities.storage.Disk.class, DiskSnapshot.class).map(disk, null);
+            DiskSnapshot diskSnapshot = getMapper(org.ovirt.engine.core.common.businessentities.storage.Disk.class, DiskSnapshot.class)
+                    .map(disk, null);
             diskSnapshot.setDisk(new Disk());
-            diskSnapshot.getDisk().setId(this.diskId.toString());
+            diskSnapshot.getDisk().setId(disk.getId().toString());
             collection.getDiskSnapshots().add(addLinks(populate(diskSnapshot, disk), Disk.class));
-            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId().toString()));
-            if (diskSnapshot.getParent() != null) {
-                diskSnapshot.getParent().setHref(buildParentHref(diskId.toString(), false));
+            diskSnapshot.setHref(buildHref(disk.getId().toString(), diskSnapshot.getId()));
+            Guid parentDiskId = disk.getParentDiskId();
+            if (parentDiskId != null) {
+                diskSnapshot.getParent().setHref(buildHref(parentDiskId.toString(),
+                        diskSnapshot.getParent().getId()));
+            } else if (diskSnapshot.getParent() != null) {
```

Comment:
You check for `diskSnapshot.getParent() != null` here, but if it possibly can be `null`, then you can fall on line 47 & 48 in `parentDiskId != null` flow?
Probably line 46 should change to `if (parentDiskId != null && diskSnapshot.getParent() != null) {`?

Reply:
> You check for `diskSnapshot.getParent() != null` here, but if it possibly can be `null`, then you can fail on line 47 & 48 in `parentDiskId != null` flow?

unless we say that if the disk has a parent disk (i.e., parentDiskId != null) then the snapshots of the disks must have parent snapshots

Reply:
In that case, it's OK, can resolve the comment :)